File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlSearchCondition.cs
using T1.Standard.IO;

namespace T1.SqlSharp.Expressions;

public class SqlSearchCondition : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.SearchCondition;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_SearchCondition(this);
    }

    public required ISqlExpression Left { get; set; }
    public LogicalOperator LogicalOperator { get; set; } = LogicalOperator.None;
    public TextSpan OperatorSpan { get; set; } = new();
    public ISqlExpression? Right { get; set; }
    public string ToSql()
    {
        var sql = new IndentStringBuilder();
        sql.Write(Left.ToSql());
        if (LogicalOperator != LogicalOperator.None)
        {
            sql.Write(" ");
            sql.Write(LogicalOperator.ToSql());
            sql.Write(" ");
            sql.Write(Right!.ToSql());
        }
        return sql.ToString();
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlSearchCondition.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlSearchCondition implements ISqlExpression {
    SqlType: SqlType = SqlType.SearchCondition;
    Span: TextSpan = new TextSpan();
    Left!: ISqlExpression;
    Right?: ISqlExpression;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_SearchCondition(this);
    }

    ToSql(): string {
        let sql = this.Left.ToSql();
        if (this.Right) {
            sql += ' ' + this.Right.ToSql();
        }
        return sql;
    }
} 